WcaOnRails/spec/controllers/teams_controller_spec.rb

```diff
@@ -180,10 +180,25 @@
       it 'cannot add overlapping membership periods for the same user'do
         member = FactoryGirl.create :user
         patch :update, id: team, team: { team_members_attributes: {"0" => { user_id: member.id, start_date: Date.today, end_date: Date.today+10, team_leader: false },
-                                                                   "1" => { user_id: member.id, start_date: Date.today+9, end_date: Date.today+20, team_leader: false }} }
+                                                                   "1" => { user_id: member.id, start_date: Date.today+9, end_date: Date.today+20, team_leader: false } } }
         invalid_team = assigns(:team)
         expect(invalid_team).to be_invalid
       end
+
+      it 'does not see an exception when tries to add another membership for the same user without start_date' do
+        member = FactoryGirl.create :user
+        expect do
+          patch :update, id: team, team: { team_members_attributes: { "0" => { user_id: member.id, start_date: Date.today+10, end_date: Date.today+5 },
+                                                                      "1" => { user_id: member.id, start_date: nil, end_date: Date.today+10 } } }
+        end.to_not raise_error
+      end
+
+      it 'does not see an exception when tries to add a member with end_date but without the start_date' do
+        member = FactoryGirl.create :user
+        expect do
+          patch :update, id: team, team: { team_members_attributes: { "0" => { user_id: member.id, start_date: nil, end_date: Date.today+5 } } }
```

Comment:
The problem with `to_not raise_error` assertions is that they are almost meaningless. You can do whatever you want (`User.delete_all`) as long as you don't raise an exception the test will be green. I'd write the test as follows:

```ruby
it 'does not see an exception when tries to add another membership for the same user without start_date' do
  member = FactoryGirl.create :user
  patch :update, id: team, team: { team_members_attributes: { "0" => { user_id: member.id, start_date: Date.today+10, end_date: Date.today+5 },
                                                              "1" => { user_id: member.id, start_date: nil, end_date: Date.today+10 } } }
  expect(team.reload.team_members.size).to eq 0
end
```

If an exception is being raised the spec will fail anyway.